File: src/main/java/aurocosh/divinefavor/common/network/message/client/spell_uses/MessageSyncFavor.java
package aurocosh.divinefavor.common.network.message.client.spell_uses;

import aurocosh.divinefavor.DivineFavor;
import aurocosh.divinefavor.common.custom_data.player.PlayerData;
import aurocosh.divinefavor.common.custom_data.player.data.favor.SpiritData;
import aurocosh.divinefavor.common.network.base.NetworkWrappedClientMessage;
import aurocosh.divinefavor.common.spirit.base.ModSpirit;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class MessageSyncFavor extends NetworkWrappedClientMessage {
    public int spiritId;
    public int favor;

    public MessageSyncFavor() {
    }

    public MessageSyncFavor(ModSpirit spirit, SpiritData spiritData) {
        this.spiritId = spirit.getId();
        favor = spiritData.getFavor(spiritId);
    }

    @Override
    @SideOnly(Side.CLIENT)
    protected void handleSafe() {
        EntityPlayer player = DivineFavor.proxy.getClientPlayer();
        SpiritData spiritData = PlayerData.get(player).getSpiritData();
        spiritData.setFavor(spiritId, favor);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/network/message/client/spirit_data/MessageSyncFavor.kt
package aurocosh.divinefavor.common.network.message.client.spirit_data

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.custom_data.player.data.favor.SpiritData
import aurocosh.divinefavor.common.lib.extensions.divinePlayerData
import aurocosh.divinefavor.common.network.base.WrappedClientMessage
import aurocosh.divinefavor.common.spirit.base.ModSpirit
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class MessageSyncFavor : WrappedClientMessage {
    var spiritId: Int = 0
    var favor: Int = 0

    constructor() {}

    constructor(spirit: ModSpirit, spiritData: SpiritData) {
        spiritId = spirit.id
        favor = spiritData.getFavor(spiritId)
    }

    @SideOnly(Side.CLIENT)
    override fun handleSafe() {
        val player = DivineFavor.proxy.clientPlayer
        player.divinePlayerData.spiritData.setFavor(spiritId, favor)
    }
}
